furmanlukasz/MatrixAutoEncoder
# Implement Streamlit Dashboard, Enhance Data Visualization, and Optimize Grid Search

This pull request introduces a new Streamlit-based frontend for our Matrix AutoEncoder Analysis Dashboard, along with several enhancements to improve data visualization and optimize the grid search functionality:


1. Streamlit Dashboard Implementation:
   - Created a new file `app.py` that serves as the main entry point for the Streamlit application.
   - Implemented a multi-page structure with navigation for different analysis steps:
     - DataFrame Selection & Visualization
     - Clustering
     - Classification
     - Statistics & AUC
     - Grid Search Heatmap
   - Added interactive elements such as file selectors, parameter inputs, and action buttons.
   - Integrated visualizations directly into the Streamlit interface.

2. Added Ridgeline Plots:
   - Implemented a new function `create_ridgeline_plot()` in `classification.py` to generate Ridgeline Plots for RQA features.
   - Updated the Streamlit app to display the Ridgeline Plots after classification.
   - These plots provide a clear visualization of the distribution of each RQA feature across different conditions.

3. Optimized Grid Search:
   - Refactored the `grid_search_heatmap()` function in `grid_search_heatmap.py` to improve performance and visualization.
   - Enhanced the heatmap visualization with more detailed and informative plots.

4. Streamlit App Improvements:
   - Implemented progress indicators and informative messages during long-running processes.
   - Added error handling.
   - Ensured responsive layout and proper display of plots and data tables.

5. Code Refactoring:
   - Improved code organization and readability across multiple files.
   - Standardized naming conventions and error handling.
   - Modularized the code to separate concerns between data processing and UI presentation.

These changes significantly enhance the usability and data analysis capabilities of our dashboard, providing users with an intuitive interface, more insightful visualizations, and faster grid search results.

To test these changes:
1. Install Streamlit and other new dependencies (listed below).
2. Run the Streamlit app using `streamlit run scripts/app/app.py`.
3. Navigate through all sections of the dashboard.
4. Perform operations like clustering, classification, and grid search.
5. Verify that all visualizations, including Ridgeline Plots and heatmaps, are generated and displayed correctly.

Dependencies added:
- streamlit
- plotly
- joypy
- seaborn (ensure latest version)
